integration-cli/docker_cli_save_load_test.go

```diff
@@ -382,3 +384,204 @@ func (s *DockerSuite) TestSaveLoadNoTag(c *check.C) {
+		exec.Command("tar", "t"))
+	c.Assert(err, checker.IsNil, check.Commentf(out))
+	c.Assert(out, checker.Contains, "refs/latest")
+	// additional test...
+	c.Assert(out, checker.Contains, "oci-layout")
+
+	// test save with 2 tags (same underlying image)
+	dockerCmd(c, "tag", "busybox:latest", "busybox:notlatest")
+	out, _, err = runCommandPipelineWithOutput(
+		exec.Command(dockerBinary, "save", "--format", "oci", "busybox:notlatest", "busybox"),
+		exec.Command("tar", "t"))
+	c.Assert(err, checker.IsNil, check.Commentf(out))
+	c.Assert(out, checker.Contains, "refs/notlatest")
+	c.Assert(out, checker.Contains, "refs/latest")
+
+	// test can save with image id
+	imageID := inspectField(c, "busybox:latest", "Id")
+	imageID = strings.Replace(imageID, "sha256:", "", -1)
+
+	out, _, err = runCommandPipelineWithOutput(
+		exec.Command(dockerBinary, "save", "--format", "oci", imageID),
+		exec.Command("tar", "t"))
+	c.Assert(err, checker.IsNil, check.Commentf(out))
+	c.Assert(out, checker.Contains, "refs/"+imageID)
+
+	// test that saving "name:tag" just includes "refs/tag" and not all the tags
+	out, _, err = runCommandPipelineWithOutput(
+		exec.Command(dockerBinary, "save", "--format", "oci", "busybox:latest"),
+		exec.Command("tar", "t"))
+	c.Assert(err, checker.IsNil, check.Commentf(out))
+	c.Assert(out, checker.Contains, "refs/")
+	c.Assert(out, checker.Contains, "refs/latest")
+	c.Assert(strings.Count(out, "refs/"), checker.Equals, 2)
+
+	// as in saving just "name"
+	dockerCmd(c, "tag", "busybox:latest", "img0")
+	dockerCmd(c, "tag", "img0:latest", "img0:notlatest")
+	dockerCmd(c, "tag", "img0:latest", "img0:another")
+	out, _, err = runCommandPipelineWithOutput(
+		exec.Command(dockerBinary, "save", "--format", "oci", "img0"),
+		exec.Command("tar", "t"))
+	c.Assert(err, checker.IsNil, check.Commentf(out))
+	c.Assert(out, checker.Contains, "refs/")
+	c.Assert(out, checker.Contains, "refs/latest")
+	c.Assert(out, checker.Contains, "refs/notlatest")
+	c.Assert(out, checker.Contains, "refs/another")
+	c.Assert(strings.Count(out, "refs/"), checker.Equals, 4)
+
+	// test --ref name:tag=reference
+	dockerCmd(c, "tag", "busybox:latest", "img0")
+	dockerCmd(c, "tag", "img0:latest", "img0:notlatest")
+	dockerCmd(c, "tag", "busybox:latest", "busybox:notlatest")
+	dockerCmd(c, "tag", "img0:latest", "img0:another")
+	out, _, err = runCommandPipelineWithOutput(
+		exec.Command(dockerBinary, "save", "--format", "oci", "--ref", "img0:notlatest=img0-notlatest-ref", "busybox:latest", "busybox:notlatest", "img0:notlatest"),
+		exec.Command("tar", "t"))
+	c.Assert(err, checker.IsNil, check.Commentf(out))
+	c.Assert(out, checker.Contains, "refs/")
+	c.Assert(out, checker.Contains, "refs/img0-notlatest-ref")
+	c.Assert(out, checker.Contains, "refs/notlatest")
+	c.Assert(out, checker.Contains, "refs/latest")
+	c.Assert(strings.Count(out, "refs/"), checker.Equals, 4)
+}
+
+func (s *DockerSuite) TestSaveOCIInternals(c *check.C) {
```

Comment:
> If OCI had some sort of verification tool to run against the saved tarball that would be nice to use, but I'd rather not put oci logic in the integration tests.

https://github.com/opencontainers/image-tools I can build that in the dev container and use it in tests to ensure everything's ok. But agree it's not really needed for the docker point of view.